browser-test/src/admin/admin_predicates.test.ts

```diff
@@ -1054,211 +1086,266 @@ test.describe('create and edit predicates', () => {
+        })
+        // Single string one of a list of strings
+        await adminPrograms.goToEditBlockEligibilityPredicatePage(
+          programName,
+          'Screen 2',
+        )
+        await adminPredicates.addPredicates({
+          questionName: 'text-question',
+          scalar: 'text',
+          operator: 'is one of',
+          value: 'blue, green',
+        })
       })
 
-      // Simple long predicate
-      await adminPrograms.goToEditBlockEligibilityPredicatePage(
-        programName,
-        'Screen 3',
-      )
-      await adminPredicates.addPredicates({
-        questionName: 'number-question-equal-to',
-        scalar: 'number',
-        operator: 'is equal to',
-        value: '42',
+      await test.step('Configure screen 3', async () => {
+        await adminQuestions.addNumberQuestion({
+          questionName: 'number-question-equal-to',
+        })
+        await adminPrograms.addProgramBlockUsingSpec(programName, {
+          name: 'Screen 3',
+          questions: [{name: 'number-question-equal-to'}],
+        })
+        // Simple long predicate
+        await adminPrograms.goToEditBlockEligibilityPredicatePage(
+          programName,
+          'Screen 3',
+        )
+        await adminPredicates.addPredicates({
+          questionName: 'number-question-equal-to',
+          scalar: 'number',
+          operator: 'is equal to',
+          value: '42',
+        })
       })
 
-      // Single long one of a list of longs
-      await adminPrograms.goToEditBlockEligibilityPredicatePage(
-        programName,
-        'Screen 4',
-      )
-      await adminPredicates.addPredicates({
-        questionName: 'number-question-one-of',
-        scalar: 'number',
-        operator: 'is one of',
-        value: '123, 456',
+      await test.step('Configure screen 4', async () => {
+        await adminQuestions.addNumberQuestion({
+          questionName: 'number-question-one-of',
+        })
+        await adminPrograms.addProgramBlockUsingSpec(programName, {
+          name: 'Screen 4',
+          questions: [{name: 'number-question-one-of'}],
+        })
+        // Single long one of a list of longs
+        await adminPrograms.goToEditBlockEligibilityPredicatePage(
+          programName,
+          'Screen 4',
+        )
+        await adminPredicates.addPredicates({
+          questionName: 'number-question-one-of',
+          scalar: 'number',
+          operator: 'is one of',
+          value: '123, 456',
+        })
       })
 
-      // Currency predicate
-      await adminPrograms.goToEditBlockEligibilityPredicatePage(
-        programName,
-        'Screen 5',
-      )
-      await adminPredicates.addPredicates({
-        questionName: 'currency-question',
-        scalar: 'currency',
-        operator: 'is greater than',
-        value: '100.01',
+      await test.step('Configure screen 5', async () => {
+        await adminQuestions.addCurrencyQuestion({
```

Comment:
comment (non-blocking):  I'm torn because I like the clarity of creating the question and adding it to the block in the same step as where we add the predicate, but I think it would improve the efficiency of the tests to create the questions and add them to the blocks in a `beforeAll` clause.  What are your thoughts about that option?

Reply:
Also https://github.com/civiform/civiform/wiki/Playwright-Feature-Migration-Guide says "The beforeAll and afterAll functions do not have access to the page object and many other fixtures" and generally discourages beforeAll... and it looks like we currently have no usages of it. I guess how much of a performance improvement do you think it might be, for it to be worth all that?

Reply:
I didn't realize that `beforeAll` was discouraged.  Good to know!  I guess it's better to stick with duplicating the question and block creation in that case.